### PR TITLE
fix relayer funding flow

### PR DESCRIPTION
**Pull request by felipemadero**
## Why this should be merged
Closes https://github.com/ava-labs/avalanche-cli/issues/2450

### CODE REVIEW

File: cmd/blockchaincmd/deploy.go [hunk 1]
```diff
@@ -1123,6 +1123,15 @@ func deployBlockchain(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	if sidecar.Sovereign {
+		ux.Logger.GreenCheckmarkToUser("L1 is successfully deployed on %s", network.Name())
+	} else {
+		ux.Logger.GreenCheckmarkToUser("Subnet is successfully deployed %s", network.Name())
+	}
+	ux.Logger.PrintToUser("")
+	ux.Logger.PrintToUser(logging.Green.Wrap("At this point your are able to interact with your L1!"))
```
Comment (arturrez) [inline cmd/blockchaincmd/deploy.go:1132]: L1 should not be used for noSoV - probably move this to `if sidecar.Sovereign`

File: cmd/blockchaincmd/deploy.go [hunk 2]
```diff
@@ -1145,58 +1154,78 @@ func deployBlockchain(cmd *cobra.Command, args []string) error {
...
-				if err != nil {
-					return err
+			icmErr = err
+			ux.Logger.RedXToUser("Interchain Messaging is not deployed due to: %v", icmErr)
+		} else {
+			ux.Logger.GreenCheckmarkToUser("ICM is successfully deployed")
+			if network.Kind != models.Local && !useLocalMachine {
+				if flag := cmd.Flags().Lookup(skipRelayerFlagName); flag != nil && !flag.Changed {
+					ux.Logger.PrintToUser("")
+					ux.Logger.PrintToUser("As Interchain Messaging was also deployed into your L1, you may")
```
Comment (arturrez) [inline cmd/blockchaincmd/deploy.go:1164]: maybe better wording:
Do you want to setup local relayer for the messages to be interchanged, as Interchain Messafing was deployed to your L1

also are we sure if we can use `L1` name and not `subnet`. maybe `blockchain` is a better term there.

File: cmd/blockchaincmd/deploy.go [hunk 3]
```diff
@@ -1145,58 +1154,78 @@ func deployBlockchain(cmd *cobra.Command, args []string) error {
...
-		return PrintSubnetInfo(blockchainName, true)
+		_ = PrintSubnetInfo(blockchainName, true)
+	}
+
+	if icmErr != nil {
+		ux.Logger.PrintToUser("")
+		ux.Logger.PrintToUser("Interchain Messaging is not deployed due to: %v", icmErr)
+		ux.Logger.PrintToUser("")
+		ux.Logger.PrintToUser("To deploy ICM later on, call `avalanche icm deploy`")
+		ux.Logger.PrintToUser("To deploy a local relayer later on, call `avalanche interchain relayer deploy`")
```
Comment (arturrez) [inline cmd/blockchaincmd/deploy.go:1220]: are you sure you want to display `avalanche interchain relayer deploy` hint for icm case ?

File: cmd/interchaincmd/relayercmd/deploy.go [hunk 4]
```diff
@@ -371,23 +377,48 @@ func CallDeploy(_ []string, flags DeployFlags, network models.Network) error {
...
+				switch {
+				case !isCChainDestination && flags.Amount != 0:
 					amountFlt = flags.Amount
-				} else {
+				case isCChainDestination && flags.CChainAmount != 0:
+					amountFlt = flags.CChainAmount
+				default:
 					amountFlt, err = app.Prompt.CaptureFloat(
-						"Amount to transfer",
+						fmt.Sprintf("Amount to transfer (available: %f)", balanceFlt),
```
Comment (arturrez) [inline cmd/interchaincmd/relayercmd/deploy.go:399]: debug or you want to use ux.PrintToUser?

File: cmd/interchaincmd/relayercmd/deploy.go [hunk 5]
```diff
@@ -371,23 +377,48 @@ func CallDeploy(_ []string, flags DeployFlags, network models.Network) error {
...
 						func(f float64) error {
 							if f <= 0 {
 								return fmt.Errorf("not positive")
```
Comment (arturrez) [inline cmd/interchaincmd/relayercmd/deploy.go:402]: nit: add value f to the fmt.Errorf

File: cmd/interchaincmd/relayercmd/deploy.go [hunk 6]
```diff
@@ -371,23 +377,48 @@ func CallDeploy(_ []string, flags DeployFlags, network models.Network) error {
...
 							}
+							if f > balanceFlt {
```
Comment (arturrez) [inline cmd/interchaincmd/relayercmd/deploy.go:404]: nit: add value f to the fmt.Errorf

File: pkg/evm/evm.go [hunk 7]
```diff
@@ -832,3 +843,11 @@ func WaitForRPC(ctx context.Context, rpcURL string) error {
 		}
 	}
 }
+
+func PrivateKeyToAddress(privateKey string) (common.Address, error) {
```
Comment (arturrez) [inline pkg/evm/evm.go:847]: nit: move this func to utils

**Review by arturrez — APPROVED** [review]

File: cmd/interchaincmd/relayercmd/deploy.go (hunk 4 above)
Comment (felipemadero) [reply cmd/interchaincmd/relayercmd/deploy.go:399]: this is just a formatting directive

**Pull request merged**